src/python/grpcio/grpc/_cython/_cygrpc/credentials.pyx.pxi

```diff
@@ -58,7 +58,10 @@ cdef int _get_metadata(void *state,
         cb(user_data, NULL, 0, status, c_error_details)
   args = context.service_url, context.method_name, callback,
   plugin = <object>state
-  plugin._stored_ctx.run(_spawn_callback_async, plugin, args)
+  if plugin._stored_ctx is not None:
```

Comment:
nit: `hasattr` might be more suitable here, or it could trigger an `AttributeError` in future for other plugin implementation.

Reply:
Can you explain how that would happen? `plugin` is an instance of [`grpc._plugin_wrapping._Plugin`](https://github.com/grpc/grpc/blob/9dabaecedabf36336c366fd62b138dd17951a357/src/python/grpcio/grpc/_plugin_wrapping.py#L67), not of [`grpc.AuthMetadataPlugin`](https://github.com/grpc/grpc/blob/9dabaecedabf36336c366fd62b138dd17951a357/src/python/grpcio/grpc/__init__.py#L613). I don't see any way for user customizations to affect the presence of this field.

Reply:
Maybe we can add a line at https://github.com/grpc/grpc/blob/9dabaecedabf36336c366fd62b138dd17951a357/src/python/grpcio/grpc/_plugin_wrapping.py#L70

```
self._stored_ctx = None
```

Just a way to show that `_stored_ctx` is a class member.

Reply:
Nevermind. You are right.